### PR TITLE
Add integration tests for NFT federation flow

### PR DESCRIPTION
**Pull request by marcciosilva**

### CODE REVIEW

File: bridge/hardhat.config.js [hunk 1]
```diff
@@ -52,7 +52,8 @@ module.exports = {
       ':SideToken',
       ':TransparentUpgradeableProxy$',
       ':NFTBridge$',
-      ':SideNFTToken'
+      ':SideNFTToken',
+      ':NFTERC721TestToken'
```
Comment (pmprete) [inline bridge/hardhat.config.js:56]: I recomend using the abi IERC721

File: bridge/package.json [hunk 2]
```diff
@@ -32,6 +32,8 @@
   "dependencies": {
     "@nomiclabs/hardhat-truffle5": "^2.0.0",
     "@nomiclabs/hardhat-web3": "^2.0.0",
+    "@thinkanddev/deploy-eip-1820-web3-rsk": "^1.0.2",
+    "@thinkanddev/hardhat-erc1820-rsk": "^0.1.2",
```
Comment (pmprete) [inline bridge/package.json:36]: i thought this was already added, weird

File: federator/integrationTest/integrationTest.js [hunk 3]
```diff
@@ -12,6 +12,10 @@ const abiSideToken = require('../../bridge/abi/SideToken.json');
 const abiAllowTokens = require('../../bridge/abi/AllowTokens.json');
 const abiMultiSig = require('../../bridge/abi/MultiSigWallet.json');
 
+const abiNFTBridge = require('../../bridge/abi/NFTBridge.json');
```
Comment (pmprete) [inline federator/integrationTest/integrationTest.js:15]: We should create a new file for the NFT integration test, something like /integrationTest/nftIntegrationTest.js

**Review by pmprete — COMMENTED** [review]
Except my comments looks good so far

File: bridge/hardhat.config.js (hunk 1 above)
Comment (marcciosilva) [reply bridge/hardhat.config.js]: never saw this, sorry! what I have available is `../../bridge/artifacts/contracts/zeppelin/token/ERC721/IERC721.sol/IERC721.json`, but throws an error when attempting to require that ABI (`Unhandled error: Error: You must provide the json interface of the contract when instantiating a contract object.`), any ideas?

Comment (marcciosilva) [reply bridge/hardhat.config.js]: the thing, however, is that I need to mint a token, and only the test token contract allows this publicly

Comment (pmprete) [reply bridge/hardhat.config.js]: never mine then, if you need to mint it, SideNFTToken is the way to go 👍

**Pull request merged**